### ISSUE TITLE
Call steps asynchronously

### ISSUE THREAD

**Issue by pimboden** [cristijora/vue-form-wizard#45]
Hello.
I'm trying to load my step-templates only when they are clicked.
Some of my steps have to load some data through ajax after loaded. 

Right now, every steps is getting the data in the "created" life cycle of vue. 

But I would like them to get the data only when the step becomes active. I there a way to do that. I haven't found anything in the documentation

Thanks for this fantastic component!

**Comment by cristijora** [comment]
Hi @pimboden 
There is no "out-of-the-box" solution for this but from what you described I think there are a couple of ways to achieve this. I will make a fiddle example. There is an `on-change` event which you could use. I will try to make a fiddle

**Comment by cristijora** [comment]
I made a fiddle based on one of the examples in the docs
https://jsfiddle.net/CristiJ/bt5dhqtf/1025/
Just go to the second step and wait for a second. After 1s, the fields will be populated as if you made a request to some server. This can be improved by making the request only once. `if(nextIndex == 1 && step1DataRequested)`
I was thinking about a better solution I could implement which would use scoped slots.
Note this is not implemented!
```html
  <tab-content title="Personal details"
                         icon="ti-user">
   <template scope="props">
         <div v-if="props.active">
            // do something here to populate data
         </div>
       </template>
  </tab-content>
```

**Comment by cristijora** [comment]
Thinking a bit more about this, I think adding scoped slots for tab content is the best approach.
You will be able to do something like this
```html
 <tab-content>
     <template scope="props">
          <step1 v-if="props.active"></step1>
     </template>
</tab-content>
```

Which will trigger `step1` to re-render each time step1 is activated which will always trigger the `created/mounted` hooks inside step1

Here is a fiddle more close to your example I think
https://jsfiddle.net/CristiJ/bt5dhqtf/1043/

**Comment by cristijora** [comment]
Released [0.6.6](https://github.com/cristijora/vue-form-wizard/releases/tag/v0.6.6)

**Comment by pimboden** [comment]
Thank you.
Your fiddle helped me resolve the issue!
